### PR TITLE
RHEL updates

### LINKED ISSUE #9856: Kernel Panic when mounting volumes on RHEL 6.5 using Docker 1.4.0, Docker 1.4.1, with latest device mapper
The problem
----

After mounting a host volume with Docker 1.4.0, Docker 1.4.1-dev, the server hangs for a few seconds and the ssh terminal session is lost with the host. Here's a Fig up session:

```
[root@pppdc9prd6dq newww]# fig up
Creating newww_npmRegistryElasticsearchData_1...
Creating newww_npmRegistryElasticsearchServer_1...
Creating newww_npmRegistryCouchdbData_1...
Creating newww_npmRegistryCouchdbServer_1...
Creating newww_npmRegistryRedisData_1...
Creating newww_npmRegistryRedisServer_1...
Write failed: Broken pipe
```

The same occurs when running the docker command manually... 

```
[root@pppdc9prd6dq newww]# docker run -ti -v $(pwd)/dev/couch/data:/usr/local/var/lib/couchdb -v $(pwd)/dev/couch/couch-docker.ini:/usr/local/etc/couchdb/local.ini busybox bash
exec: "bash": executable file not found in $PATHFATA[0000] Error response from daemon: Cannot start container e8e53a826d1602e1ca5363109db401856963f47313f2d25a8e6a2d9b3e718c3a: exec: "bash": executable file not found in $PATH 
[root@pppdc9prd6dq newww]# docker run -ti -v $(pwd)/dev/couch/data:/usr/local/var/lib/couchdb -v $(pwd)/dev/couch/couch-docker.ini:/usr/local/etc/couchdb/local.ini busybox sh
/ # ls -la
total 56
drwxr-xr-x   17 root     root          4096 Dec 30 18:51 .
drwxr-xr-x   17 root     root          4096 Dec 30 18:51 ..
-rwxr-x---    1 root     root             0 Dec 30 18:51 .dockerenv
-rwxr-x---    1 root     root             0 Dec 30 18:51 .dockerinit
drwxrwxr-x    2 root     root          4096 May 22  2014 bin
drwxr-xr-x    5 root     root           380 Dec 30 18:51 dev
drwxr-xr-x    6 root     root          4096 Dec 30 18:51 etc
drwxrwxr-x    4 root     root          4096 May 22  2014 home
drwxrwxr-x    2 root     root          4096 May 22  2014 lib
lrwxrwxrwx    1 root     root             3 May 22  2014 lib64 -> lib
lrwxrwxrwx    1 root     root            11 May 22  2014 linuxrc -> bin/busybox
drwxrwxr-x    2 root     root          4096 Feb 27  2014 media
drwxrwxr-x    2 root     root          4096 Feb 27  2014 mnt
drwxrwxr-x    2 root     root          4096 Feb 27  2014 opt
dr-xr-xr-x  158 root     root             0 Dec 30 18:51 proc
drwx------    2 root     root          4096 Dec 30 18:51 root
lrwxrwxrwx    1 root     root             3 Feb 27  2014 run -> tmp
drwxr-xr-x    2 root     root          4096 May 22  2014 sbin
drwxr-xr-x   13 root     root             0 Dec 30 18:28 sys
drwxrwxrwt    3 root     root          4096 May 22  2014 tmp
drwxrwxr-x    7 root     root          4096 Dec 30 18:51 usr
drwxrwxr-x    4 root     root          4096 May 22  2014 var
/ # ls -la /usr/local/var/lib/couchdb/
total 8
drwxr-x---    2 root     root          4096 Dec 30 09:50 .
drwxr-x---    3 root     root          4096 Dec 30 18:51 ..
/ # ls -la /usr/local/etc/couchdb/local.ini 
-rw-------    1 root     root          1359 Dec 30 07:14 /usr/local/etc/couchdb/local.ini
/ # exit
Write failed: Broken pipe
```

I cannot ssh to the host until I restart the server... ssh to the host becomes unavailable... 

```
mdesales@SDGl121456EB ~ (mac) $ ssh npm.corp.intuit.net 
ssh: connect to host npm.corp.intuit.net port 22: Operation timed out
```

Reproducing
-----

First, I installed the latest Docker 1.4.0 from yum EPEL, and later built Docker 1.4.1 on a Ubuntu machine and scp the latest version to the host. Still the same problem occurs. Everything started with the following error:

http://stackoverflow.com/questions/27216473/docker-1-3-fails-to-start-on-rhel6-5

So I upgraded the device mapper library to the latest version:

```
$ cd /etc/yum.repos.d 
$ wget public-yum.oracle.com/public-yum-ol6.repo 
$ wget public-yum.oracle.com/RPM-GPG-KEY-oracle-ol6 -O /etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
$ yum update device-mapper
```

After that, I was able to restart Docker 1.4.0 or Docker 1.4.1-dev without a problem... 

```
[root@pppdc9prd6dq mdesales]# docker --version
Docker version 1.4.1-dev, build a444643
```

I can list images without a problem and manipulate them using Fig:

```
[root@pppdc9prd6dq newww]# fig up
Creating newww_npmRegistryElasticsearchData_1...
Cannot start container 62c57ef8830000496a8c6dfeba637dc8976de47d82431f3df701c6bdccfe6d5d: Cannot find child for /newww_npmRegistryElasticsearchData_1
[root@pppdc9prd6dq newww]# fig rm
Going to remove newww_npmRegistryRedisServer_1, newww_npmRegistryRedisData_1, newww_npmRegistryCouchdbServer_1, newww_npmRegistryCouchdbData_1, newww_npmRegistryElasticsearchServer_1
Are you sure? [yN] y
Removing newww_npmRegistryElasticsearchServer_1...
Removing newww_npmRegistryCouchdbData_1...
Removing newww_npmRegistryCouchdbServer_1...
Removing newww_npmRegistryRedisData_1...
Removing newww_npmRegistryRedisServer_1...
```

After that, I tried to "fig up" again and the same problem occurs...

```
[root@pppdc9prd6dq newww]# fig up
Creating newww_npmRegistryElasticsearchData_1...
Creating newww_npmRegistryElasticsearchServer_1...
Creating newww_npmRegistryCouchdbData_1...
Creating newww_npmRegistryCouchdbServer_1...
Creating newww_npmRegistryRedisData_1...
Creating newww_npmRegistryRedisServer_1...
Write failed: Broken pipe
```

### PR DESCRIPTION
**Pull request by vbatts**
- update version 6.5 -> 6.6
- bump kernel version

Closes #9856 

/cc @LK4D4 @unclejack @thaJeztah @tiborvass

### CODE REVIEW

**Comment by tiborvass** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM, thanks! Perhaps you can add a comment on the linked issue to inform people arriving there that the (kernel) requirements have changed. (For those arriving there through Google)

**Comment by vbatts** [comment]
@thaJeztah I thought about that, though wasn't sure if we wanted the docs to continue growing with links to issues...

**Comment by vbatts** [comment]
@thaJeztah added

**Comment by thaJeztah** [comment]
@vbatts oh sorry if I wasn't clear, I meant just a comment on #9856 here on the issue tracker. Some people are lazy and just start commenting on an issue without reading all comments, so a "final" comment hopefully prevents that.

**Comment by vbatts** [comment]
oh right. I thought the inverse, but for the same reason. If I just left a
comment on the issue, then folks may continue commenting on the closed
issue and miss the "final" resolution. But if the link is in the docs, that
is really, very final.

This happens a lot with devicemapper issues. We make the "final" comment,
but folks just ignore the fix/solution and continue to comment, then it's
lost. :-\

Should I remove this link from the docs?

**Comment by thaJeztah** [comment]
> Should I remove this link from the docs?

Hm, not sure it adds much to the docs, but just leave it for now until @moxiegirl has had a look, she might have a more firm preference either way (and the commits also have to be squashed, so you're not done yet 😇)

File: docs/sources/installation/rhel.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ page_keywords: Docker, Docker documentation, requirements, linux, rhel
 Docker is supported on the following versions of RHEL:
 
 - [*Red Hat Enterprise Linux 7 (64-bit)*](#red-hat-enterprise-linux-7-installation)
-- [*Red Hat Enterprise Linux 6.5 (64-bit)*](#red-hat-enterprise-linux-6.5-installation) or later
+- [*Red Hat Enterprise Linux 6.6 (64-bit)*](#red-hat-enterprise-linux-6.6-installation) or later
```
Comment (moxiegirl) [inline docs/sources/installation/rhel.md]: This link doesn't work as written.  You need to change `6.6` to `66`

**Comment by moxiegirl** [comment]
I believe officially we only require the 3.10 kernel or higher which is what 7 is running. I've been told by the team we only support kernel 2.6.32 on RedHat so for 6.6 you have that covered. 

Just that l link fix should do it.

File: docs/sources/installation/rhel.md (hunk 1 above)
Comment (thaJeztah) [inline docs/sources/installation/rhel.md]: Oh! Well spotted!

**Comment by thaJeztah** [comment]
> I believe officially we only require the 3.10 kernel or higher which is what 7 is running. I've been told by the team we only support kernel 2.6.32 on RedHat so for 6.6 you have that covered.

Correct, IIRC the minimum kernel is 3.13. The RedHat / CentOS distros are an exception to this because (as mentioned) they back-port changes that are needed for Docker to work; Kernel 2.6.x on other distros is *not* supported because those lack those changes.

**Comment by SvenDowideit** [comment]
LGTM, once @vbatts fixes the link - easy way to test is to run `make docs` in your branch.

**Comment by vbatts** [comment]
fixed the link

**Comment by moxiegirl** [comment]
@SvenDowideit Thanks, I know that. That's how I found the break.

**Comment by moxiegirl** [comment]
@vbatts everything LGTM. Thank you for the contribution.

**Pull request merged**